mlflow/utils/environment.py

```diff
@@ -189,6 +192,31 @@ def _is_iterable(x):
         )
 
 
+_INFER_PIP_REQUIREMENTS_FALLBACK_MESSAGE = (
+    "Encountered an unexpected error while inferring pip requirements (model URI: %s, flavor: %s)"
+)
+
+
+def infer_pip_requirements(model_uri, flavor, fallback=None):
```

Comment:
`infer_pip_requirements` significantly increase the `flavors` test execution time (by 10 ~ 15 min).

Reply:
I don't think we should do anything non-obvious to change the behavior under the hood. Can we remove `sklearn`, `gluon`, etc from our model flavor tests as part of this work to keep CI time low? This will add some time to cross version tests, but I think that's okay.

Reply:
Agree!